### ISSUE TITLE
Forwarding of Local routes is not consulted with the routing table

### ISSUE THREAD

**Issue by ManiAm** [inet-framework/inet#368]
When a router receives a datagram with destination IPv4 address of a directly-connected network, then it will deliver the datagram directly and it never forwards it. In other words, the router does not consult its routing table for local addresses due to the following line:

https://github.com/inet-framework/inet/blob/62ac2bc230d2dd9f4fa466d79f71a08dd79e0425/src/inet/networklayer/ipv4/Ipv4.cc#L348

I was thinking that the routing table is searched from top to bottom and the first match (longest prefix match) is used for forwarding. But apparently, local addresses are treated differently. This causes a bug when you add a static route to a router in order to forward datagrams differently. Configuration Step5 in the following location demonstrates the issue.

https://github.com/ManiAm/inet/blob/INETex/examples_INETex/static_dynamic_routing/omnetpp.ini

But you need to change the configurator config to the following:

```
*.configurator.config = xml("<config> \
								<interface hosts='**' address='10.x.x.x' netmask='255.x.x.x'/> \
                                <route hosts='host*' destination='*' netmask='0.0.0.0' interface='eth0' /> \
	                            <route hosts='router0' destination='10.0.0.17' netmask='255.255.255.255' gateway='10.0.0.22' interface='eth2' metric='12' /> \
                             </config>")
```

The above config adds a static route into router0 as highlighted with red rectangle.

![2018-07-23_15-09-20](https://user-images.githubusercontent.com/5473998/43105738-cdbf192e-8e8a-11e8-9bdc-30c427dbf685.png)

Although in router0's routing table, the first route explicitly says that any datagram with destination 10.0.0.017 should be forwarded to eth2, router0 will not respect this static route and since 10.0.0.17 is a local address, it will be delivered locally which is a bug!

**Comment by levy** [comment]
Let me rephrase what you are saying to make sure that I understand correctly. When a packet has a destination address that is assigned to a router's network interface (local network interface address), the router still has to look for matching routes in its routing table and potentially skip local delivery altogether. Do I understand correctly?

**Comment by ManiAm** [comment]
>> Let me rephrase what you are saying to make sure that I understand correctly. When a packet has a destination address that is assigned to a router's network interface (local network interface address), the router still has to look for matching routes in its routing table and potentially skip local delivery altogether. Do I understand correctly?

Yes.
